### PR TITLE
some more python3 adaptations

### PR DESCRIPTION
**Pull request by fchapoton**
Progress, but blocked by twisted, where twisted.mail is not available in python3 (even in the most recent version of twisted)

### CODE REVIEW

**Comment by kcrisman** [comment]
There wasn't ever really that much use of email other than probably account recovery, and #120 was one of the major places this was added in a useful way.  In `sagenb/notebook/notebook.py` you might even be able to just remove the logger for email by more or less reverting that one.

Interestingly, what is https://twistedmatrix.com/trac/ticket/8770 ?  Is it in a version of Twisted we have access to?

Also, see https://stackoverflow.com/questions/37299924/send-email-with-twisted-on-python3 which suggests maybe it would be possible to just replace with [smtplib](https://docs.python.org/3/library/smtplib.html#module-smtplib).

**Comment by fchapoton** [comment]
I thought that I managed to install twisted 18.4.0 using pip in sage shell, but I got

In [1]: import twisted
In [2]: twisted.version
Out[2]: Version('twisted', 16, 3, 0)

inside ipython called from sage shell

**Comment by embray** [comment]
I'm amused by the oxymoron "blocked by twisted".

**Comment by embray** [comment]
Should we just upgrade the twisted version in Sage?  Is think twisted is only included for sagenb in the first place.

**Comment by kcrisman** [comment]
> Should we just upgrade the twisted version in Sage? Is think twisted is
> only included for sagenb in the first place.
>
>
I think that is pretty likely.

**Comment by dimpase** [comment]
twisted is used in few doctests, see https://trac.sagemath.org/ticket/8785

File: sagenb/misc/misc.py [hunk 1]
```diff
@@ -21,6 +21,14 @@
 
 from pkg_resources import resource_filename
 
+try:
+    CRE = ConnectionRefusedError
+    PYTHON_VERSION = 3
```
Comment (dimpase) [inline sagenb/misc/misc.py]: this seems to be a weird way to detect Python version, no?

**Comment by dimpase** [comment]
have you pulled the latest master into your branch?

**Comment by fchapoton** [comment]
Yes, I worked on top of the latest master: cloned the master, pulled my branch on top of that and work there.

**Comment by dimpase** [comment]
> I worked on top of the latest master: cloned the master, pulled my branch on top of that and work there.

but the master is no longer the one you used, I think. I believe there should be no changes in sphinxify.py in your pull request. How about doing a git pull, and then updating the pull request?

**Comment by fchapoton** [comment]
I do not understand what happened. I was using the very latest master. Anyway, I have rebased and my branch looks better now.

**Comment by dimpase** [comment]
> I do not understand what happened. I was using the very latest master.

it was latest 19 days ago; there is no automatic update of the base of the fork, the diff shown is against the base branch you used originally, not its update.

(this is different to what we do on trac, where the updates to the beta change the diffs)

**Pull request merged**